.github/workflows/main.yaml

```diff
@@ -0,0 +1,161 @@
+name: Quality Checks
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the code
+      - name: Checkout
+        uses: actions/checkout@v4
+      # Install Python
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      # Install dependencies
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest pylint
+          for FILE in $(find . -name 'requirements.txt'); do
+            pip install -r $FILE
+          done
+
+      # Create .util
+      - name: Create .util
+        run: |
+          [ ! -d .util ] && mkdir -p .util || true
+
+      # Run pytest
+      - name: Run tests
+        if: '!cancelled()'
```

Comment:
What is the use of this if statement? It occurs multiple times

Reply:
It a check that runs the step only if the run hasn't been cancelled.